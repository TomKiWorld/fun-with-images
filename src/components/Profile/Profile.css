.images-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 90%;
  max-width: 75rem;
  margin: 0 auto;
}

.remove-button {
  background-color: #E900A3;
  color: white;
  padding: 0.5rem;
  border: none;
  border-radius: 0.5rem;
}