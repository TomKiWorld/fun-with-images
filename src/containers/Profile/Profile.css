.images-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 90%;
  max-width: 75rem;
  margin: 0 auto;
}

.cta-button {
  margin: 0.5rem;
  background-color: #E900A3;
  color: white;
  padding: 0.5rem;
  border: none;
  border-radius: 0.5rem;
}

.profile-model {
  background-color: rgba(255, 255, 255, 0.7);
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
}

.close-modal {
  font-size: 3rem;
  position: absolute;
  right: 1rem;
  top: 0;
  cursor: pointer;
}

.close-modal:hover {
  color: grey;
}

.profile-container {
  position: relative;
  background-color: #FFF;
  width: 22rem;
}

.profile-user-name {
  display: inline-block;
  margin-left: 2rem;
}

.profile-info {
  margin: 1rem auto;
  border-bottom: 0.1rem solid rgba(92, 218, 218, 0.7);
}

.profile-buttons {
  display: flex;
  justify-content: space-around;
}

.profile-button {
  width: 6rem;
}

.delete-profile-button {
  display: block;
  margin: 0.5rem auto;
}

.error-message {
  text-align: center;
}
