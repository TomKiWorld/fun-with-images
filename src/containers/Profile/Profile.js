import React, { Component } from 'react';
import { DATABASE } from '../../constants';

import './Profile.css';
import ImageList from '../../components/ImageList/ImageList';
import Preloader from '../../components/Preloader/Preloader';
import ProfileGreet from '../../components/ProfileGreet/ProfileGreet';

/**
 * User Profile
 * - Showes summary of how many times the user used the site
 * - Shows a list of previously submitted images
 * - Offers the option to resubmit past images
 * - Allowes the user to delete his/ hers profile from the database
 * 
 * Required props:
 * - user => User from Redux state
 * - onRouteChange => Function to change route
 * - onResubmit => Function to handle resubmission of images
 */
class Profile extends Component {
  constructor() {
    super();
    this.state = {
      images: []
    }
  }

  // Fetch user submitted images from dtatebase
  componentDidMount() {
    this.getProfileImages();
  }

  // Get profile images
  getProfileImages = () => {
    return fetch(`${DATABASE}/profile-images/${this.props.user.id}`)
    .then(response => response.json())
    .then(data => {
      if(data.length) {
        this.setState({images: data})
      }
    })
    .catch(err => console.log('Could not fetch images'));
  }

  // Resubmit an image from the profile page
  onImageResubmit = (url) => {
    this.props.onResubmitInput(url);
    this.props.onRouteChange('home');
    this.props.onImageChange('');
    this.props.onImageError('');
    this.props.onResubmitImageUrl(true);
  }

  // Delete the profile only in case profile is not 'Visitor'
  onSubmitRemoval = () => {
    fetch(`${DATABASE}/profile-removal/${this.props.user.id}`, {
      method: 'post',
      headers: {'Content-type': 'application/json'},
      body: JSON.stringify({
        email: this.props.user.email,
        id: this.props.user.id
      })
    })
    .then(response => response.json())
    .then(data => {
      alert(data);
    })
    .then(() => this.props.onRouteChange('signin'))
    .catch(err => alert('Unable to remove profile'));
  }

  showRemoveButton = (id) => {
    if (id !== 1) {
      return (
        <button 
        className='remove-button'
        onClick={this.onSubmitRemoval}
        >Delete your profile</button> 
      )
    } else {
      return <button className='remove-button'>You can not remove this profile</button>
    }
  }

  render() {
    const { user } = this.props;
    let images = user.entries > 0 ? 
      <Preloader /> : 
      <p>Go ahead and submit <span role='img' aria-label='Smirking Face'>😏</span></p>;
    if (this.state.images.length) {
      images = 
        <ImageList 
          images={this.state.images} 
          onResubmit={this.onImageResubmit}/>
    }

    return(
      <article className='profile pa4 mb4'>
        <h1>Hi <span className='capitalize'>{user.name}</span></h1>
        <ProfileGreet entries={user.entries} />
        {this.showRemoveButton(user.id)}
        <section>
          {images}
        </section>
      </article>
    )
  }
}
  
export default Profile;
