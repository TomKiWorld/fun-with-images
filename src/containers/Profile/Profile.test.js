import { shallow } from 'enzyme';
import React from 'react';
import Profile from './Profile';

describe('Profile Component', () => {
  const mockProps = {
    user: {
      id: 2,
      name: 'John',
      email: 'john@gmail.com',
      entries: 0
    }
  }
  const wrapper= shallow(<Profile {...mockProps} />)

  it('Expect to render Profile Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
