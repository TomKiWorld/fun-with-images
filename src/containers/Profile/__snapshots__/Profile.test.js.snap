// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component Expect to render Profile Component 1`] = `ShallowWrapper {}`;
